eng/common/TestResources/New-TestResources.ps1

```diff
@@ -119,6 +119,26 @@ function MergeHashes([hashtable] $source, [psvariable] $dest) {
     }
 }
 
+function BuildBicepFile([System.IO.FileSystemInfo] $file) {
+    if (!(Get-Command bicep -ErrorAction Ignore)) {
```

Comment:
I don't think this is an appropriate check. According to https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/deploy-powershell, Az 5.6.0 and newer support bicep. Why not just check if Az is that version or newer and let the cmdlets do their job?

```powershell
if (!(Get-Module -FullyQualifiedName @{ModuleName='Az'; ModuleVersion='5.6.0'} -ListAvailable)) {
  throw 'No'
}
```

Reply:
This way, too, you don't introduce yet another dependency beyond the cmdlets.

Reply:
We decided to call the tool directly because the rest of the deployment script depends on having a arm template json file for things like parsing parameters, etc, so we decided to call bicep build up front and keep the rest of the script the same.

Reply:
Add some context Wes' comment: while `az bicep build` is a supported command for the `az` cli, there is no equivalent `bicep build` command to compile to ARM templates in the `Az` cmdlet (not yet at least).

This check is really intended for local usage of `New-TestResources.ps1` only, since the azure pipelines images include the Bicep command in the default image (except for MacOS, as I discovered today, so this is a separate problem that needs solving).

Reply:
1. Is there a tracking issue to eventually switch over when they do support it?
2. Could you update the message to recommend, for example, running `az bicep install`?

This script is meant as much for manual runs as for automation: that was the primary goal of these scripts - to provide a consistent means of provisioning test resources. If users are trying to run tests and don't have bicep installed, it's not clear now how they should remediate.

Reply:
Also, for future context re arm template json: it's not just to keep the rest of the script the same, but to avoid having to write custom parsing for bicep file parameters.